Snowflake.Data/Core/Tools/UnixOperations.cs

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Mono.Unix;
+using Mono.Unix.Native;
+using System.IO;
+
+namespace Snowflake.Data.Core.Tools
+{
+    internal class UnixOperations
+    {
+        public static readonly UnixOperations Instance = new UnixOperations();
+
+        private UnixFileInfo _unixFileInfo;
+        private UnixDirectoryInfo _unixDirInfo;
+
+        public virtual void SetDirInfo(string path)
+        {
+            _unixDirInfo = new UnixDirectoryInfo(path);
+        }
+
+        public virtual void CreateDirectoryWithPermissions(string path, FilePermissions permissions)
+        {
+            string subPath = Path.GetDirectoryName(path);
```

Comment:
so let's say that `path = "/home/user/easy_logging_dir/dotnet"`. Let's say that `/home/user/easy_logging_dir` exists and `dotnet` subfolder does not exist. In this case subPath will be `/home/user/easy_logging_dir/dotnet` and it all be created normally without special permissions?

For me it would be better if you would create the directory `/home/user/easy_logging_dir` in the EasyLoggingStarter class just as under Windows, and use UnixOperations just to create the `dotnet` part.

Reply:
Usually the `GetDirectoryName` function gets the path except the last part but I didn't take into account the Unix/Linux path format and so it still returns the whole path instead

Will change this